### ISSUE TITLE
Default queue gets distributed to all queues

### ISSUE THREAD

**Issue by m1racoli** [apache/airflow#149]
Hi,
we are currently running Airflow 1.1.1 and observed a weird behaviour regarding the default-queue setting for Celery and don't know if it's intended behaviour or a bug.

If an operators queue happens to be set to the same queue as the default-queue set up in the Celery configuration, then the task gets submitted by the scheduler to each queue currently registered in Redis.

We expected that the default queue is only crucial for the worker to listen to, when it is not assigned to a specific queue.

**Comment by mistercrunch** [comment]
You said "the task gets submitted by the scheduler to each queue currently registered in Redis", do you mean exactly that, or that it gets submitted once to all **workers** regardless of their queue settings? I'd be surprised if a task instance gets submitted multiple times to different queues.

You may want to upgrade to `1.2` to rule that out. You'll have to run a single `airflow initdb` after upgrading.

**Comment by mistercrunch** [comment]
I re-read the Airflow code around `default_queue` and this celery docs and everything seem ok as far as I can tell...:
http://celery.readthedocs.org/en/latest/userguide/routing.html#routing-automatic

**Comment by jomar83** [comment]
Thanks for the quick response.

To clarify, we have a dedicated worker listening to a specific queue (airflow worker -q "specialqueue").
Now  the issue here is that every task that goes into the default queue also ends up in the "specialqueue" (yes, it isn't just handled by the worker, it really ends up in the redis list for the specialqueue). The way to resolve this is to set different default queues for both workers. That's however not the best solution in case we would like the worker to handle specific tasks as well as the ones targeting the default queue.

Can you explain that behaviour?

**Comment by mistercrunch** [comment]
So if you go to the "Task Details" view in the UI for a task exhibiting this issue, what queue do you see? "default"?

Are you still on `1.1`? Can you upgrade to `1.2`? And confirm it's still happening?

We are using the queuing feature in production at Airbnb and it's been reliable, I haven't seen any tasks go to the wrong worker...

**Comment by m1racoli** [comment]
I'm currently looking into it and trying to reproduce the behaviour on a local installation. I'll get back to you.

**Comment by m1racoli** [comment]
I tried to reproduce the behaviour on a separate setup on my local machine, but I couldn't. Still it occurred on our server.

I suspected malformed data in Redis. Thus we flushed all Celery data in Redis. After that I could not reproduce the behaviour on our server as well.

It seemed like that at the end temporarily async default_queue configurations among the workers might have caused the problem, which was the case for some time. Only bringing the config in sync didn't help, but flushing all the data in Redis did.

p.s. we are still running version 1.1.1, which at the end wasn't the cause of the problem. But we're anyways working right now on bringing our system to 1.2.0.

Best
